features/images.feature

```diff
@@ -0,0 +1,40 @@
+@images
+Feature: OpenStack Image management
+  As an administrator
+  I want to make sure that OpenStack Image management component
+  is working correctly
+
+  @create_kvm
+  Scenario: Creation of KVM image
+    Given KVM source image exists
+    And glance image does not exist
+    When I create new glance image based on jeos
+    Then this image has non-empty ID
+    And its ID can be used to show the image info
+    And the status of the image is active
+    When I delete the image
```

Comment:
I think I do.

You are proposing that Background ensures that the image does not exist yet. That makes some sense, yes, but it doesn't mean that the state of the system is same after the tests are run, because the images are not deleted after the creation.

Unless we are using After hooks, yes, where we can delete them. But the discussion you are pointing to surely doesn't show any agreement if we should do it in the After hooks or not.

In general, I still find current approach more readable and understandable, although yours might be more pure and principally better.